lib/did_you_mean/word_collection.rb

```diff
@@ -14,8 +15,10 @@ def each(&block) words.each(&block); end
     def similar_to(target_word)
       target_word = target_word.to_s
       threshold   = threshold(target_word)
-
-      select {|word| Levenshtein.distance(word.to_s, target_word) <= threshold }
+      map { |word| [Levenshtein.distance(word.to_s, target_word), word] }
+        .select { |distance, _| distance <= threshold }
+        .sort { |a, b| a[0] <=> b[0] }
+        .map { |_, word| word }
```

Comment:
I like what you wanted to do here, but creating an array for each word could be slower than several distance re-calculations after `#select`. And again, don't include multiple changes in one pull request.